examples/src/main/kotlin/io/ethers/examples/batchrequests/Multicall.kt

```diff
@@ -0,0 +1,69 @@
+package io.ethers.examples.batchrequests
+
+import io.ethers.abi.call.Multicall3
+import io.ethers.abi.call.aggregate
+import io.ethers.core.types.Address
+import io.ethers.core.types.BlockId
+import io.ethers.examples.gen.ERC20
+import io.ethers.providers.Provider
+import kotlinx.cli.ArgParser
+import kotlinx.cli.ArgType
+import kotlinx.cli.default
+import java.math.BigInteger
+
+/**
+ * Multicall example demonstrates how to aggregate multiple contract calls into Multicall3.
+ */
+class Multicall(
+    private val rpcUrl: String,
+    private val tokens: List<Address>,
+) {
+    // Init provider
+    private val provider = Provider.fromUrl(rpcUrl).unwrap()
+
+    fun run() {
+        val aggregate = Multicall3.aggregate(
+            tokens.map { ERC20(provider, it).name() }.aggregate(),
+            tokens.map { ERC20(provider, it).symbol() }.aggregate(),
+            tokens.map { ERC20(provider, it).decimals() }.aggregate(),
+        ).call(BlockId.LATEST).sendAwait().unwrap()
+
+        val nameResults = aggregate.getAsAggregation<String>(0).unwrapOrNull()
+        val symbolResults = aggregate.getAsAggregation<String>(1).unwrapOrNull()
+        val decimalsResults = aggregate.getAsAggregation<BigInteger>(2).unwrapOrNull()
```

Comment:
Let's add an explanation that each aggregated call result is returned in the same order the calls were defined.